src/cockpit/channels/packages.py

```diff
@@ -53,18 +53,11 @@ def do_done(self) -> None:
         options = self.options
 
         try:
-            if options.get('method') != 'GET':
+            if get_str(options, 'method') != 'GET':
                 raise ValueError(f'Unsupported HTTP method {options["method"]}')
 
-            path = options.get('path')
-            if not isinstance(path, str) or not path.startswith('/'):
-                raise ValueError(f'Unsupported HTTP method {options["method"]}')
-
-            headers = options.get('headers')
-            if not isinstance(headers, dict) or not all(isinstance(value, str) for value in headers.values()):
-                raise ValueError(f'Unsupported HTTP method {options["method"]}')
-
-            document = packages.load_path(path, headers)
+            headers = get_dict(options, 'headers')
```

Comment:
This is now losing the type check that all headers are strings. Same bogus copy-pasta error message of course.

Reply:
This, together with the requirement to parse the `requires` field of manifest is pushing me towards wanting to add a sort of `get_a{ss}` helper.  Maybe `get_strdict` or something?

I think it doesn't matter a lot anyway, though, since we will catch the errors when we look up the individual headers.  Thoughts?

Reply:
get_strdict() sounds nice. But if non-string values are caught somewhere else, that's good enough for me.

Reply:
`parse_accept_language()` does this:

```python
        accept_language = get_str(headers, 'Accept-Language', None)
```

and (now updated) we also do it for the CSP-rewriting stuff in the same function just below:

```python
                    protocol = get_str(headers, 'X-Forwarded-Proto')
                    host = get_str(headers, 'X-Forwarded-Host')
```
```

Reply:
Thanks -- that seems fine! Feel free to resolve. I actually think that's easier and more flexible than get_strdict().